presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -527,16 +529,19 @@ protected Node visitShowFunctions(ShowFunctions node, Void context)
             ImmutableList.Builder<Expression> rows = ImmutableList.builder();
             for (SqlFunction function : metadata.listFunctions(session)) {
                 Signature signature = function.getSignature();
+                boolean builtIn = signature.getName().getFunctionNamespace().equals(DEFAULT_NAMESPACE);
                 rows.add(row(
-                        signature.getName().getFunctionNamespace().equals(DEFAULT_NAMESPACE) ?
-                                new StringLiteral(signature.getNameSuffix()) :
-                                new StringLiteral(signature.getName().toString()),
+                        builtIn ? new StringLiteral(signature.getNameSuffix()) : new StringLiteral(signature.getName().toString()),
                         new StringLiteral(signature.getReturnType().toString()),
                         new StringLiteral(Joiner.on(", ").join(signature.getArgumentTypes())),
                         new StringLiteral(getFunctionType(function)),
                         function.isDeterministic() ? TRUE_LITERAL : FALSE_LITERAL,
                         new StringLiteral(nullToEmpty(function.getDescription())),
-                        signature.isVariableArity() ? TRUE_LITERAL : FALSE_LITERAL));
+                        signature.isVariableArity() ? TRUE_LITERAL : FALSE_LITERAL,
+                        builtIn ? TRUE_LITERAL : FALSE_LITERAL,
+                        function instanceof SqlInvokedFunction
+                                ? new StringLiteral(((SqlInvokedFunction) function).getRoutineCharacteristics().getLanguage().name().toLowerCase(ENGLISH))
+                                : new NullLiteral()));
```

Comment:
Maybe you also want to use `nullToEmpty` as in description?

Reply:
I think it make sense to have empty description, but for a more enum-like metadata, maybe we want to keep it as `null`?

Reply:
I prefer null when programmatically read it but it hurts my eyes 😂. Maybe we should say the language is JAVA for builtin functions then? Your call.

Reply:
I would prefer empty string then, as a form of emptiness.